File: __python__/binToDec.py
def twoPower(x):
	res = 1
	for i in range(x): res *= 2
	return res

def binToDec(s):
	n = len(s)-1
	dec_val = 0

	for i in range(n, -1, -1):
		dec_val += (int(s[n-i])*twoPower(i))

	return dec_val


def main():
	s = "1011"
	print(binToDec(s))  # 11

if __name__ == "__main__":
	main()

File: __cpp__/algorithms/binToDec.cpp
#include <iostream>
#include <string>
using namespace std;

int two_power(int x) {
	int res = 1;
	for(int i=0; i<x; i++) {
		res = res * 2;
	}
	return res;
}

int binary_to_decimal(string s) {
	int n = s.length()-1;
	int decimal_value = 0;
	
	for(int i=n; i>-1; i--) {
		decimal_value = decimal_value + (int(s[n-i]) * two_power(i));
	}
	
	return decimal_value;
}

int main() {
	string s = "1101";
	cout << binary_to_decimal(s);
	
	return 0;
}
